src/main/java/org/spongepowered/api/map/MapView.java

```diff
@@ -0,0 +1,245 @@
+     */
+    void sendUpdate(Collection<Player> players, Vector2i min, Vector2i max);
+
+
+    /**
+     * Returns if the vanilla behavior for map updates (slowly updating a few
+     * rows) at a time is being used.
+     *
+     * @return True if vanilla behavior is being used, false otherwise
+     */
+    boolean doesAutomaticUpdates();
+
+    /**
+     * Sets if vanilla behavior for map updates (slowing updating a few rows
+     * at a time) is being used. The only way a map will update if set to false
+     * is if plugins call on of the {@link #sendUpdate} methods manually, which
+     * could be useful for maps that should constantly update.
+     *
+     * @param doesAutomaticUpdates True if vanilla behavior should be used,
+     *        false otherwise
+     */
+    void setDoesAutomaticUpdates(boolean doesAutomaticUpdates);
+
+    /**
+     * Returns if before using any {@link MapRenderer}s the {@link MapCanvas} should
+     * have the vanilla map copied into it so plugins can do overlay rendering on maps
+     * without implementing vanilla map behavior themselves.
+     *
+     * @return True if vanilla rendering is used as a base, false otherwise
+     */
+    boolean usesDefaultRenderer();
+
+    /**
+     * Sets if vanilla's map rendering should be added to the {@link MapCanvas} before
+     * plugin {@link MapRenderer}s are run to allow overlaying.
+     *
+     * @param usesDefaultRenderer True to enable vanilla rendering as a base, false
+     *        otherwise
+     */
+    void setUsesDefaultRenderer(boolean usesDefaultRenderer);
+
+    /**
+     * Adds a renderer to this map as the top layer.
+     *
+     * @param renderer The renderer to add
+     */
+    void addRenderer(MapRenderer renderer);
+
+    /**
+     * Adds a renderer to this map as the base layer.
+     *
+     * @param renderer The renderer to add
+     */
+    void addRendererBase(MapRenderer renderer);
+
+    /**
+     * Inserts a renderer in the map at the specific index, shifting existing
+     * renderers around it.
+     *
+     * @param index The index to insert the renderer at
+     * @param renderer The renderer to insert
+     * @throws IndexOutOfBoundsException If the provided index is out of bounds
+     */
+    void insertRenderer(int index, MapRenderer renderer);
+
+    /**
+     * Removes the specified renderer from the rendering list.
+     *
+     * @param renderer The renderer to remove
+     * @return True if the renderer was successful to be removed, false
+     *         otherwise
+     */
+    boolean removeRenderer(MapRenderer renderer);
+
+    /**
+     * Removes the {@link MapRenderer} at the specified index from the
+     * renderers on this map
+     *
+     * @param index The index of the renderer to remove
+     * @return True The renderer removed
+     * @throws IndexOutOfBoundsException If the provided index is out of bounds
+     */
+    MapRenderer removeRenderer(int index);
```

Comment:
Why would you return `Optional`? AFAIK the list of renderers cannot contain null and removing a random / none existing index is just plain wrong so the AIOOBException is right IMO as it is more precise than an IllegalArgumentException.